src/uri.rs

```diff
@@ -315,6 +318,22 @@ impl Display for Uri {
     }
 }
 
+#[cfg(feature = "compat")]
+impl From<http_types::Uri> for Uri {
+    fn from(uri: http_types::Uri) -> Uri {
+        uri.to_string().parse()
```

Comment:
The `to_string` is unfortunate, but it doesn't look like there's a good way around it, as `hyper::Uri` provides `as_ref` for `&str`, requiring the holding of 1 contiguous buffer.

... Hm, though, `http::Uri` is backed by `Bytes`, just like `hyper::Uri`. I haven't looked to remind myself, but I wonder if there's a combination of methods that can be called to try to prevent a full alloc and copy?

Reply:
I'll see if I can come up with something. Maybe there's a feature we can land in `http` to enable a more performant conversion here?

Reply:
I've explored this a bit:

- Converting from `hyper::Uri` to `http::Uri` can be done without copying the underlying data. In fact, I've included this in my changes to the branch.

- Converting  from `http::Uri` to `hyper::Uri` is not easily possible, however, since `http` "chops" the `Bytes` into different parts. If allocations here really become a problem, I guess we could consider changing `http::Uri` to keep a reference to the "top-level" `Bytes` so that we could implement `http::Uri::as_bytes` or something similar.